poetry/core/packages/file_dependency.py

```diff
@@ -63,8 +63,8 @@ def full_path(self):
     def is_file(self):
         return True
 
-    def hash(self):
-        h = hashlib.sha256()
+    def hash(self, name="sha256"):
+        h = hashlib.new(name)
```

Comment:
All the docs say the named constructors are faster and should be preferred, and this doesn't do anything to address the issue.  It can be safely dropped from the PR.